bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/internal/javasound/JavaSoundAudioSource.java

```diff
@@ -63,13 +58,7 @@ public JavaSoundAudioSource() {
     private TargetDataLine initMicrophone(javax.sound.sampled.AudioFormat format) throws AudioException {
         try {
             TargetDataLine microphone = AudioSystem.getTargetDataLine(format);
-
-            DataLine.Info info = new DataLine.Info(TargetDataLine.class, format);
```

Comment:
Looking at this code now, I wonder whether there wasn't a good reason to have it.
I personally develop on macOS as well, so the simpler code might just work out.
But I remember that Kelly Davis developed on Linux and the Linux audio system behaved quite differently to macOS.
Before merging this, I'd hence think it should be tested on Linux (and possibly Windows) as well. Do you have any chance to do so?

Reply:
Sure, I can perform those tests tomorrow.
I'll ping you with the results.

Reply:
You were right. There are some limitations for open multiple TargetDataLine for same format, but under Windows I found comments reporting same problem for other projects. I also discover there are some problems on windows with the porcupineKS add-on, I'll look at them, so the tests where really meaningful. On linux (Ubuntu 20.04.3 LTS) everything works right. I didn't found any comments reporting the problem on linux.
I'll update the pr to keep sharing the microphone under windows, but closing it correctly when no more streams depend on it.

Reply:
Alright, sounds good - thanks for the elaborate testing!

Reply:
I have updated the implementation to behave different on windows.